### PR TITLE
Bug at Documentation: Wrong Error mentioned

### PR DESCRIPTION
**Pull request by chetkhatri**
What were proposed in this Pull Request ?
* iex> length([1, [2], 3]) = 3
above syntax won't give below error 
** (CompileError) iex:1: illegal pattern
but it gives ** (CompileError) iex:1: cannot invoke remote function :erlang.length/1 inside match

so it has been manually verified at shell and update the documentation

### CODE REVIEW

File: crash-course.markdown [hunk 1]
```diff
@@ -634,13 +634,19 @@ Anonymous functions are first-class values, so they can be passed as arguments t
...
+```
 
-lists:map(fun math:square/1, [1, 2, 3]).
-%=> [1, 4, 9]
+```erl
+Eshell V5.9  (abort with ^G)
+1> c(math).
+{ok,math}
+2> lists:map(fun math:square/1, [1, 2, 3]).
+[1,4,9]
```
Comment (whatyouhide) [inline crash-course.markdown]: Is this change needed/relevant for this PR?

Comment (chetkhatri) [inline crash-course.markdown]: Looks, predecessor PR got merged ! let me check
